azurerm/resource_arm_traffic_manager_endpoint.go

```diff
@@ -281,5 +365,46 @@ func getArmTrafficManagerEndpointProperties(d *schema.ResourceData) *trafficmana
 		endpointProps.MinChildEndpoints = &mci64
 	}
 
+	templist := d.Get("subnet").([]interface{})
+	subnetMappings := make([]trafficmanager.EndpointPropertiesSubnetsItem, 0)
+	for _, subnetOld := range templist {
+		subnetOld := subnetOld.(map[string]interface{})
+		subnetFirst := subnetOld["first"].(string)
+		subnetLast := subnetOld["last"].(string)
+		subnetScope := int32(subnetOld["scope"].(int))
+		var subnetNew trafficmanager.EndpointPropertiesSubnetsItem
+		if subnetScope == 0 && subnetFirst != "0.0.0.0" {
+			subnetNew = trafficmanager.EndpointPropertiesSubnetsItem{
+				First: &subnetFirst,
+				Last:  &subnetLast,
+			}
+		} else {
+			subnetNew = trafficmanager.EndpointPropertiesSubnetsItem{
+				First: &subnetFirst,
+				Scope: &subnetScope,
+			}
+		}
+		subnetMappings = append(subnetMappings, subnetNew)
+	}
+	if len(subnetMappings) > 0 {
+		endpointProps.Subnets = &subnetMappings
+	}
+
+	templist = d.Get("custom_header").([]interface{})
+	headerMappings := make([]trafficmanager.EndpointPropertiesCustomHeadersItem, 0)
+	for _, headerOld := range templist {
```

Comment:
We could simplify this
```suggestion
	for _, headerOld := range d.Get("custom_header").([]interface{}) {
```

Reply:
Also could we not use `headerOld` as the meaning is not clear. I would recommend header

Reply:
Sure